### PR TITLE
FEATURE: Force admin sidebar for all admins in `admin_sidebar_enabled_groups` and handle legacy "hamburger dropdown" in this mode

### PR DESCRIPTION
**Pull request by martin-brennan**
Some sites are still on the legacy "hamburger dropdown"
navigation_menu setting. In this case to avoid confusion,
we want to show both the sidebar icon and the header dropdown
hamburger when visiting the admin portal. Otherwise, the
hamburger switches sides from right to left for admins
and takes on different behaviour.

The hamburger in this case _only_ shows the main panel, not
other sidebar panels like the admin one.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/header/sidebar-toggle.gjs [hunk 1]
```diff
@@ -30,7 +30,7 @@ export default class SidebarToggle extends Component {
         aria-controls="d-sidebar"
         {{on "click" this.toggleWithBlur}}
       >
-        {{icon "bars"}}
+        {{icon "discourse-sidebar"}}
```
Comment (martin-brennan) [inline app/assets/javascripts/discourse/app/components/header/sidebar-toggle.gjs:33]: Should be conditional, only show discourse-sidebar with the weird combo

File: app/assets/javascripts/discourse/app/components/header/icons.gjs [hunk 2]
```diff
@@ -27,9 +27,25 @@ export function clearExtraHeaderIcons() {
 export default class Icons extends Component {
   @service site;
   @service currentUser;
+  @service siteSettings;
   @service header;
   @service search;
 
+  get showHamburger() {
+    // NOTE: In this scenario, we are forcing the sidebar on admin users,
```
Comment (martin-brennan) [inline app/assets/javascripts/discourse/app/components/header/icons.gjs:35]: This weird combo is used in a few places, might want to move this condition into sidebar state

**Review by Drenmi — APPROVED** [review]
LGTM! 🚀 

Part of me would like to find a shorter name for `adminSidebarAllowedWithLegacyNavigationMenu`, but I don't think it's critical. 🙂

**Comment by martin-brennan** [comment]
> Part of me would like to find a shorter name for adminSidebarAllowedWithLegacyNavigationMenu, but I don't think it's critical. 🙂

Yeah I think it's fine mainly because it signals that, yes, this is kind of a screwed up thing to have to support :joy:

**Pull request merged**